homeassistant/components/media_player/__init__.py

```diff
@@ -308,6 +319,22 @@ def media_player_service_handler(service):
                                descriptions.get(service),
                                schema=MEDIA_PLAYER_SCHEMA)
 
+    def turn_on_service(service):
+        """Set specified volume on the media player."""
+        source = service.data.get(ATTR_INPUT_SOURCE)
+
+        for player in component.extract_from_service(service):
+            player.turn_on()
+            if source:
+                player.select_source(source)
```

Comment:
Why add kwargs to onkyo player `turn_on` method if you're calling `select_source` method separately?